modules/javafx.graphics/src/main/native-glass/mac/GlassPasteboard.m

```diff
@@ -775,12 +775,28 @@ static inline jobject createUTF(JNIEnv *env, NSString *data) {
                 }
             }
 
-            // http://developer.apple.com/library/mac/#documentation/cocoa/Conceptual/PasteboardGuide106/Articles/pbCustom.html
-            [pasteboard writeObjects:objects];
+            // We perform a Drag-n-Drop only when our pasteboard is a DnD pasteboard
+            // and if drag delegate was set. The latter one is set when DnD operation
+            // came from a NSView (ex. mouse drag).
+            if ([[pasteboard name] isEqualToString:NSDragPboard] && [GlassDragSource isDelegateSet])
+            {
+                // DnD requires separate NSDragging* calls to work since macOS 10.7
+                // convert NSPasteboardItem-s array to NSDraggingItem-s
+                NSMutableArray<NSDraggingItem*> *dItems = [NSMutableArray<NSDraggingItem*> arrayWithCapacity:itemCount];
+                for (NSPasteboardItem* i in objects)
+                {
+                    [dItems addObject:[[NSDraggingItem alloc] initWithPasteboardWriter:i]];
```

Comment:
I think you need to call `autorelease` on the allocated `NSDraggingItem` (unless I missed seeing a release later on).